pkg/monitortests/etcd/etcdloganalyzer/monitortest.go

```diff
@@ -138,11 +143,25 @@ func (*etcdLogAnalyzer) ConstructComputedIntervals(ctx context.Context, starting
 		newInterval = nil
 	}
 
-	return ret, nil
+	w.constructed = ret
+	return w.constructed, nil
 }
 
-func (*etcdLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
-	return nil, nil
+func (w *etcdLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
```

Comment:
What is the difference between the value in the new field `constructed` and the value passed here as the parameter `finalIntervals`? Is the new field necessary?

Reply:
Agree with Ben - these are the same intervals:

https://github.com/Elbehery/origin/blob/909a424355bc8039e1af981c5c1f67c695107018/pkg/monitor/monitor.go#L103

https://github.com/Elbehery/origin/blob/909a424355bc8039e1af981c5c1f67c695107018/pkg/monitor/monitor.go#L113

https://github.com/Elbehery/origin/blob/909a424355bc8039e1af981c5c1f67c695107018/pkg/monitor/monitor.go#L123-L126

Reply:
I've updated the test to avoid using `constructed` and instead filter `finalIntervals` PTAL - https://github.com/openshift/origin/commit/ec983ec8992a4874d621a7d53425781253a26b6e

The way to test this is to run the monitor and stop two master machines (one of them have to be etcd leader) so that it could not elect a new one for some time

Reply:
actually will add this manually, the commit has two parents, cherrypicking will be mess :D

Reply:
Feel free to merge it with your commit afterwards

Reply:
updated 👍🏽